dateparser/date.py

```diff
@@ -304,6 +314,11 @@ def get_date_data(self, date_string, date_formats=None):
         TODO: Timezone issues
 
         """
+        global default_language_loader
+        if settings.SKIP_TOKENS != self._skip_tokens:
+            default_language_loader = LanguageDataLoader()
+            self = DateDataParser(*self._default_args)
```

Comment:
You understand that this is happening every time you call `get_date_data` because you updating local variable `self` and not the instance itself, right?
Also, as for now we don't expose our settings interface to user, let's assume it's not changing and move this check to `__init__` method, getting rid of `global` in the same time. Instead, create a class method `get_language_loader` and use class attribute to lazy instantiate default language loader.